src/HealthChecks.AzureServiceBus/AzureServiceBusHealthCheck.cs

```diff
@@ -7,21 +6,18 @@ namespace HealthChecks.AzureServiceBus;
 
 public abstract class AzureServiceBusHealthCheck<TOptions> where TOptions : AzureServiceBusHealthCheckOptions
 {
-    protected static readonly ConcurrentDictionary<string, ServiceBusClient> ClientConnections = new();
-
-    protected static readonly ConcurrentDictionary<string, ServiceBusAdministrationClient> ManagementClientConnections = new();
-
-    protected static readonly ConcurrentDictionary<string, ServiceBusReceiver> ServiceBusReceivers = new();
-
     protected TOptions Options { get; }
 
+    private readonly ServiceBusClientProvider _clientProvider;
+
     protected string Prefix => Options.ConnectionString ?? Options.FullyQualifiedNamespace!;
 
     protected abstract string ConnectionKey { get; }
```

Comment:
All implementation return `Prefix` from `ConnectionKey`. Let's make ConnectionKey virtual:
` protected virtual string ConnectionKey => Options.ConnectionString ?? Options.FullyQualifiedNamespace!;` and mark `Prefix` obsolete.

Reply:
On it. Was also not a fan of keeping `Prefix` to make as little change as possible as the name does not make sense.